tests/testthat/test-vroom_write.R

```diff
@@ -202,6 +202,22 @@ test_that("vroom_write equals the same thing as vroom_format", {
   expect_equal(readChar(tf, file.info(tf)$size), vroom_format(df))
 })
 
+test_that("vroom_format handles empty data frames", {
+  df <- data.frame()
+  expect_equal(vroom_format(df), "")
+
+  df <- data.frame(a = 1:2, b = 2:3)
+  df <- df[0, ]
+  expect_equal(vroom_format(df), "a\tb\n")
+
+  df <- tibble::tibble()
+  t <- tempfile(fileext = ".csv")
+  on.exit(unlink(t))
+
+  vroom::vroom_write(df, t)
+  expect_equal(vroom::vroom(t, col_types = list()), df)
+})
+
```

Comment:
Tests look good.

I propose splitting into two tests, because one is about `vroom_format()` and the other is about `vroom_write()`.

Like this:

``` r
test_that("vroom_format handles empty data frames", {
  df <- data.frame()
  expect_equal(vroom_format(df), "")

  df <- data.frame(a = 1:2, b = 2:3)
  df <- df[0, ]
  expect_equal(vroom_format(df), "a\tb\n")
})

test_that("vroom_write() / vroom_read() roundtrips an empty data frame", {
  df <- tibble::tibble()
  t <- tempfile(fileext = ".csv")
  on.exit(unlink(t))

  vroom_write(df, t)
  expect_equal(vroom(t, show_col_types = FALSE), df)
})
```

Two other changes I made:

* I'm not qualifying anything with `vroom::`. I'm not sure why we see that elsewhere in this file, but I'm assuming it's inessential weirdness.
* I'm using `show_col_types = FALSE` instead of `col_types = list()` (which I realize is used elsewhere), just because it feels like it does *less* and that seems like a good idea. If the goal is just to suppress col spec printing, let's do as little as possible to suppress the printing.

Basically, these are both things I don't see a need to change elsewhere, but we might as well write new tests in the way that makes the most sense to us.

Reply:
I like your suggestions and I updated the tests and add a NEWS bullet. If everything else looks good I think this is ready.